### PR TITLE
Added a new ship rebelStarship in new faction : rebel with all required files.

### PR DESCRIPTION
**Pull request by Graviton48**
# Description
> This PR adds the rebel module it the game. This module has the following ship:
1. rebelStarship
2. rebelMedium
3. rebelLarge
4. rebelCapital
5. rebelStation
6.  Also added  -  Projectile rebelX
7. Added a Readme to the rebel module.

# Testing
> Testable by pulling the module and running the game.

# Outstanding Work
- [ ] If there is work still outstanding, please include it here
- [ ] if there is nothing outstanding, remove this section.

# Notes
>Got settings.ini to normal, added fixed gun and ammo, made the icon white, changed the file name to module.info, I'll send a link to the original site from where I've taken the sprites.
-added rebelLarge with new gun"Destroyer".
-added rebelCapital to the game.
-added rebelStation, also a new projectile "rebelX". Also,got wave projectile in projectilesConfig.json to 
 normal.
-added a ReadMe to the rebel module.

### Pre Pull Request Checklist:

- [ ] Code has been scanned with [SonarLint](http://www.sonarlint.org/intellij/)
- [ ] Methods have been annotated with @ Nullable and @ Nonnull ([Read more](https://github.com/MovingBlocks/DestinationSol/wiki/@Nullable-&-@Nonnull-Quickstart))
- [ ] There are no errors present in the project
- [ ] Code has been formatted and indented
- [ ] Methods have appropriate Javadoc ([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by vampcat** [comment]
@GooeyHub: ok to test

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
I've approved the task for now, but before you work on the next ship, I would like you to fix the collision box of this ship. Basically, the wings should be a "solid" object that don't collide with anything. However, right now they don't collide, probably because you've not set the boundary properly.
If you're unsure how to do this, ask Suraj, he can help you.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Graviton48** [comment]
I sure will. But I have already made the new ship, so after submitting this new ship I will fix the collision box.Thank you.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: modules/core/assets/configs/projectilesConfig.json [hunk 1]
```diff
@@ -302,23 +302,23 @@
...
-        "collisionEffect": {
-            "effectFile": "core:explFire",
-            "size": 0.8,
-            "tex": "fire",
-            "tint": "20 255 20"
-        },
-
-        "collisionSound": "core:waveCollision"
-    }
+"wave": {
```
Comment (vampcat) [inline modules/core/assets/configs/projectilesConfig.json]: Is there any reason you decided to make this change?

File: modules/rebel/assets/configs/projectilesConfig.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "laser1": {
```
Comment (vampcat) [inline modules/rebel/assets/configs/projectilesConfig.json]: `laser1`
There you are, in the middle of space, fighting enemies and looting planets, when suddenly you see your greatest enemy, the LittleKnight from Deepspace faction. You pull out your blaster and shoot.. `laser1`s? That kills the entire mood. Be a bit more imaginative with your names.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/resources/settings.ini [hunk 3]
```diff
@@ -4,7 +4,7 @@ fullscreen=false
 controlType=1
 sfxVol=1.0
 musicVol=0.25
-canSellEquippedItems=false
```
Comment (vampcat) [inline engine/src/main/resources/settings.ini]: Any reason for changing this?

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/resources/settings.ini (hunk 3 above)
Comment (Graviton48) [reply engine/src/main/resources/settings.ini]: I changed it to true deliberately ,in-game.

Comment (vampcat) [reply engine/src/main/resources/settings.ini]: Actually, I can agree with this change - the player shouldn't be able to sell equipped items. Leave it be for now.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**